core/src/main/java/org/infinispan/remoting/transport/Transport.java

```diff
@@ -178,4 +181,23 @@
     */
    Set<String> getSitesView();
 
+   <T> CompletionStage<T> invokeCommand(Address target, ReplicableCommand command,
+                                        ResponseCollector<T> collector, DeliverOrder deliverOrder,
+                                        long timeout, TimeUnit unit);
+
+   <T> CompletionStage<T> invokeCommand(Collection<Address> targets, ReplicableCommand command,
```

Comment:
shouldn't be `CompletableFuture`? 
I think the `RpcManager` and the `AsyncInterceptor` use the `CompletableFuture`